### DISCUSSION TITLE
New feature suggestions

### DISCUSSION THREAD

**Discussion by AncientMystic** [open-webui/open-webui#2555] [General]
**Workspace:**
System prompt option in workspace similar to prompts but for system prompts not bound to modelfiles specifically

Characters / agents option in workspace similar to a modelfile builder ui where you can add agent / character name, avatar, system prompt, memories (similar to the prompt suggestion option) and allow agents to be tagged in chats either via singular messages or the whole chat, to be used with any model loaded and this would also introduce the possibility of multiple agents / characters being used simultaneously similar to a multi agent prompt and have their avatar appear in the chat when the user addresses them 

Options in workspace to add system wide memories for all user accounts 

**Memories:**
Import / export option for memories 

Rebuild option to reload memories after changing the embedding model

Database option for memories to switch between different active sets of memories for different purposes

**Documents:**
Rebuild option to reload documents after changing the embedding model 

**Loras:**
Better lora support 

Ui option to pull lora models

Ui option to load lora adapter with model  

**Accounts:**
Database options in accounts to import/export and switch between / create new databases for rag, memory, chats, etc

**Chat/UI:**
Chat folders / groups, the ability to organise chats into folders / groups 

Custom chat avatars 

More theme options, colors, background photos / wallpapers

Possibly the ability to search / add modelfiles and prompts from the openwebui.com site inside the UI 

**Plugin support:**
Plugins and addons options 

Chatbot options there are chatbots for ollama using email, discord, telegram, element and others it would be nice to be able to link an account with open webui and then have the AI models chat in groups / conversations within these applications 

WatchDog email notifications, possibility to add an email account to notify admin if ollama is down or other problems occur 

**Cloudflare / Remote access:**
Cloudflare tunnel option to allow remote access by friends 

Any of these things i think would be great additions to Open-WebUi

**Comment by justinh-rahb** [comment]
Converted this to discussion because it's kinda all over the place. We'd prefer that issues be somewhat constrained to a single actionable item.

As for your suggestions, there's a few I'll comment on as they've come up/been handled previously:

> Characters

- #467 (there's some other related issues but this one is still open)

> Loras

AFAIK Ollama doesn't support layering a LoRA on top of another model, this would likely need to wait until we begin the work of integrating `llama.cpp` directly into the backend.

> Accounts/database options

I'm not totally sure what you mean, but there's already places for the admin to download a database. There's no import functionality for that, and implementing it would be awkward and a lot of work when all one needs to do is create a new volume and put the exported database in it, or copy it to the `backend/data/` folder otherwise.

> Chat folders

- #1182 (see this existing issue)

> Custom chat avatars

Possible already using Modelfiles, and may also be doable in soon-to-come update that will add ability for admins to modify model names, descriptions, capabilities exposed, etc. see:
- #2505 
- #665 

> More theme options, colors, etc.

Not that we're opposed, but these are pretty low priority as features, so absent a PR from a new contributor I don't think any current ones are working on this for the time being.

> Plugin support

Just launched this the other day: https://github.com/open-webui/pipelines/

> Chatbots

Nothing stopping you from creating one as an external project and using our API, we even have some ideas of a couple we've got on the backburner for our own use in the project.

> WatchDog email notifications

Would need email-sending support to be added first, there is a PR that's still being worked on that will hopefully be adding account recovery emails (as a first offering) but I've asked that it be developed such that we can reuse the email-sending parts of it for other uses:

- #2003 

Alternatively, we could leverage webhooks as we already do for new-user signup notifications. There's been some discussion about additional event types we could have:

- #1240 

> Cloudflare

We've already got this covered... or any other method you want to use for connectivity, it can be done outside of our image, not sure why everyone wants us to build things in that we'd need to maintain forever when it can be bolted on with another Docker container in your stack, for example:

https://docs.openwebui.com/tutorial/sso
